src/util/environment.ts

```diff
@@ -0,0 +1,329 @@
+import packageJson from "../../package.json";
+
+let os: any;
+try {
+  os = require("node:os");
+} catch (_) {
+  os = undefined;
+}
+
+/**
+ * Predicate to determine if we're executing in a Node environment or not
+ * @internal
+ */
+export const isNode = (): boolean => {
+  if (typeof process !== "undefined" && process.release?.name === "node") {
+    try {
+      require("node:http2");
+      return true;
+    } catch (_) {
+      return false;
+    }
+  }
+  return false;
+};
+
+/**
+ * Function to put all of the environment details together.
+ * @internal
+ */
+export const getDriverEnv = (): string => {
+  const driverEnv = {
+    driver: ["javascript", packageJson.version].join("-"),
+    env: "unknown",
+    os: "unknown",
+    runtime: "unknown",
+  };
+
+  let isServiceWorker = false;
+  try {
+    // @ts-expect-error Cannot find name 'ServiceWorkerGlobalScope'.ts(2304)
+    isServiceWorker = global instanceof ServiceWorkerGlobalScope;
+  } catch (_) {
+    // okay if this fails
+  }
+
+  if (isNode()) {
+    driverEnv.runtime = ["nodejs", process.version].join("-");
+    driverEnv.env = getNodeRuntimeEnv();
+    driverEnv.os = [os.platform(), os.release()].join("-");
+  } else if (isServiceWorker) {
+    driverEnv.runtime = "Service Worker";
```

Comment:
Which cases does this condition cover, and are we sure that os won't be available/useful, or would it be better to try to obtain them?

Reply:
I don't think the executing context (global scope) for a service worker has anything else useful in it. I think it's just the [ServiceWorkerGlobalScope](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope).

Reply:
I see [ServiceWorkerGlobalScope.navigator](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/navigator), though. We should be able to get the browser info from that.

I'll work on that 👍🏻

Reply:
Cool! updated so ServiceWorker should send the same fields as a browser.